fe/fe-core/src/main/java/com/starrocks/catalog/PhysicalPartitionImpl.java

```diff
@@ -92,8 +95,9 @@ public class PhysicalPartitionImpl extends MetaObject implements PhysicalPartiti
 
     private volatile long minRetainVersion = 0;
 
-    public PhysicalPartitionImpl(long id, long parentId, long sharedGroupId, MaterializedIndex baseIndex) {
+    public PhysicalPartitionImpl(long id, String name, long parentId, long sharedGroupId, MaterializedIndex baseIndex) {
         this.id = id;
+        this.name = name;
```

Comment:
shall check if the name is null? and turn to "" for null?

Reply:
if the name is null, it will be generated using <partition_name>_<id> in method `Partition::gsonPostProcess()` which will called after deserialized from json.

Reply:
gsonPostProcess() just ensures its load from json and fill the name if is null. but the constructor and the setName() never promises the name is not null, so there is an implicit rule here that the user who uses this class should never provide null value or setName(null), otherwise NPE may occur.

Is it possible to remove this implicit rule by converting the null to "" for every possible interface? that's the fundamental benefit from a OOP language.

Reply:
Why do this implicitly for users ? If a use provide null value or setName(null), either it's a bug or it's what user want. If it's a bug, the NPE will help user debug. If it's what user want, the use must handle null value. If a user want a "", he can provide "" explicitly. In either case, I think it's a bad idea to do the implicit conversion.

Reply:
because the referring place of getName() doesn't check the nullable. You can either check the nullable of getName() whenever it get used or make sure the getName() doesn't return null.

It's irresponsible to say it's the caller's duty to set null or not. it is how you design the behavior of the class and make it well behaved.

Reply:
From now, it's expected that name is not null. Its value is either a string that user provided or the default value `<partition_name>_<id>` if user not provide or system upgrade from an old version. If NPE occurs, there may be a bug.